api/agent/hybrid/client.go

```diff
@@ -117,26 +117,26 @@ func (cl *client) Finish(ctx context.Context, c *models.Call, r io.Reader, async
 	return err
 }
 
-func (cl *client) GetApp(ctx context.Context, appName string) (*models.App, error) {
+func (cl *client) GetApp(ctx context.Context, app *models.App) (*models.App, error) {
 	span, ctx := opentracing.StartSpanFromContext(ctx, "hybrid_client_get_app")
 	defer span.Finish()
 
 	var a struct {
 		A models.App `json:"app"`
 	}
-	err := cl.do(ctx, nil, &a, "GET", "apps", appName)
+	err := cl.do(ctx, nil, &a, "GET", "apps", app.Name)
```

Comment:
we want to use id here for the same reason as in the cache

Reply:
If we want runner use app IDs instead of names we'd have to have to define at least 2 more API handlers that would treat `:app` as app id:
```go
	if s.nodeType != ServerTypeAPI {
		runner := engine.Group("/r")
		runner.Use(appNameCheck)
		runner.Any("/:app", s.handleFunctionCall)
		runner.Any("/:app/*route", s.handleFunctionCall)
	}
```
If so, then i can do so.

Reply:
we could probably look up by name and check the id here for a match for getting app, for route though it's not that easy. for hybrid yea maybe we just need to add 2 more handlers for lookup by id like you say

Reply:
Could the same handlers check ID or name?

Reply:
only issue would be if an app name looks like an id then it will be searched as one, not that this is much of a practical concern in core, but if adding authentication it could be a concern. if we explicitly have an endpoint to look up by id this simply isn't an issue.

Reply:
Good point, makes sense.